ui/component/publishFile/view.jsx

```diff
@@ -63,39 +73,99 @@ function PublishFile(props: Props) {
+
+  const PROCESSING_MB_PER_SECOND = 0.5;
+  const MINUTES_THRESHOLD = 30;
+  const HOURS_THRESHOLD = MINUTES_THRESHOLD * 60;
+  const MARKDOWN_FILE_EXTENSIONS = ['txt', 'md', 'markdown'];
+  const sizeInMB = Number(size) / 1000000;
+  const secondsToProcess = sizeInMB / PROCESSING_MB_PER_SECOND;
+
+  const fileSelectorModes = [
+    { label: __('Select Replay'), actionName: SOURCE_SELECT, icon: ICONS.MENU },
+    { label: __('Upload'), actionName: SOURCE_UPLOAD, icon: ICONS.PUBLISH },
+    { label: __('Not Yet'), actionName: SOURCE_NONE, icon: ICONS.REMOVE },
+  ];
+
+  const hasLivestreamData = livestreamData && Boolean(livestreamData.length);
+  const showSourceSelector = isLivestreamClaim && hasLivestreamData;
+
   const ffmpegAvail = ffmpegStatus.available;
   const [oversized, setOversized] = useState(false);
   const [currentFile, setCurrentFile] = useState(null);
   const [currentFileType, setCurrentFileType] = useState(null);
   const [optimizeAvail, setOptimizeAvail] = useState(false);
   const [userOptimize, setUserOptimize] = usePersistedState('publish-file-user-optimize', false);
+  const [fileSelectSource, setFileSelectSource] = useState(
+    IS_WEB && showSourceSelector ? SOURCE_SELECT : SOURCE_UPLOAD
+  );
+  // const [showFileUpdate, setShowFileUpdate] = useState(false);
+  const [selectedFileIndex, setSelectedFileIndex] = useState(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const totalPages =
+    hasLivestreamData && livestreamData.length > PAGE_SIZE ? Math.ceil(livestreamData.length / PAGE_SIZE) : 1;
 
-  const RECOMMENDED_BITRATE = 6000000;
-  const TV_PUBLISH_SIZE_LIMIT: number = 4294967296;
-  const TV_PUBLISH_SIZE_LIMIT_STR_GB = '4';
   const UPLOAD_SIZE_MESSAGE = __(
     '%SITE_NAME% uploads are limited to %limit% GB. Download the app for unrestricted publishing.',
     { SITE_NAME, limit: TV_PUBLISH_SIZE_LIMIT_STR_GB }
   );
-  const PROCESSING_MB_PER_SECOND = 0.5;
-  const MINUTES_THRESHOLD = 30;
-  const HOURS_THRESHOLD = MINUTES_THRESHOLD * 60;
-  const MARKDOWN_FILE_EXTENSIONS = ['txt', 'md', 'markdown'];
-
-  const sizeInMB = Number(size) / 1000000;
-  const secondsToProcess = sizeInMB / PROCESSING_MB_PER_SECOND;
 
   // Reset filePath if publish mode changed
   useEffect(() => {
     if (mode === PUBLISH_MODES.POST) {
       if (currentFileType !== 'text/markdown' && !isStillEditing) {
         updatePublishForm({ filePath: '' });
       }
+    } else if (mode === PUBLISH_MODES.LIVESTREAM) {
+      updatePublishForm({ filePath: '' });
     }
   }, [currentFileType, mode, isStillEditing, updatePublishForm]);
 
+  // set default file source to select if necessary
+  useEffect(() => {
+    if (hasLivestreamData && isLivestreamClaim) {
+      setFileSelectSource(SOURCE_SELECT);
+    } else if (isLivestreamClaim) {
+      setFileSelectSource(SOURCE_NONE);
+    }
+  }, [hasLivestreamData, isLivestreamClaim, setFileSelectSource]);
+
+  const normalizeUrlForProtocol = (url) => {
```

Comment:
Why do we need this?

Reply:
In at least one case, the urls come back inconsistently with or without https:// . I'm not sure how many are wrong.